temporalcli/commandsmd/commands.md

```diff
@@ -1,221 +1,537 @@
+Fail an Activity, marking it as having encountered an error. Specify the
+Activity and Workflow IDs:
 
-`temporal activity fail --activity-id=MyActivityId --workflow-id=MyWorkflowId`
+```
+temporal activity fail \
+    --activity-id YourActivityId \
+    --workflow-id YourWorkflowId
+```
 
 #### Options
 
-* `--activity-id` (string) - The Activity to be failed. Required.
-* `--detail` (string) - JSON data describing reason for failing the Activity.
-* `--identity` (string) - Identity of user submitting this request.
-* `--reason` (string) - Reason for failing the Activity.
+* `--activity-id` (string) -
+  Activity ID to fail.
+  Required.
+* `--detail` (string) -
+  Reason for failing the Activity (JSON).
+* `--identity` (string) -
+  Identity of the user submitting this request.
+* `--reason` (string) -
+  Reason for failing the Activity.
 
 Includes options set for [workflow reference](#options-set-for-workflow-reference).
 
+### temporal batch: Manage batch jobs
 
-### temporal batch: Manage Batch Jobs
+A batch job executes a command on multiple Workflow Executions at once. Use
+an SQL-like `--query` to select which Workflow Executions to include:
 
-Batch commands change multiple Workflow Executions.
+```
+temporal batch workflow cancel \
+  --query 'ExecutionStatus = "Running" AND WorkflowType="YourWorkflow"' \
+  --reason "Testing"
+```
+
+The `batch` keyword is optional when using `--query`. The following command
+is functionally identical to the previous one:
+
+```
+temporal workflow cancel \
+  --query 'ExecutionStatus = "Running" AND WorkflowType="YourWorkflow"' \
+  --reason "Testing"
+```
+
++----------------------------------------------------------------------------+
+| Query Quick Reference:                                                     |
+|                                                                            |
+| Composition:                                                               |
+| - Values: strings (use single or double quotes), numbers, Booleans         |
+| - Comparison: '=', '!=', '>', '>=', '<', '<='                              |
+| - Operations: 'IN array', 'BETWEEN value AND VALUE', 'STARTS_WITH string', |
+|   'IS NULL', 'IS NOT NULL', 'expr AND expr', 'expr OR expr', '( expr )'    |
+| - Array: '( comma-separated-values )'                                      |
+|                                                                            |
+| Info:                                                                      |
+| - Wrap the attribute with backticks if it contains characters not in       |
+|  `[a-zA-Z0-9]`.                                                            |
+| - `STARTS_WITH` is only available for Keyword search attributes.           |
++----------------------------------------------------------------------------+
+
+Visit https://docs.temporal.io/visibility to read more about Search Attributes
+and Query creation. See `temporal batch --help` for a quick reference.
```

Comment:
The quick reference is right above, no? We probably don't need to call it out here.

Reply:
Yeah, this is one of the two I mentioned yesterday. Fixing.